src/VirtoCommerce.NotificationsModule.Data/Services/NotificationSearchService.cs

```diff
@@ -25,68 +25,86 @@ public NotificationSearchService(Func<INotificationRepository> repositoryFactory
+                result.Results = result.Results.Concat(transientNotificationsQuery)
+                    .AsQueryable()
+                    .OrderBySortInfos(BuildSortExpression(criteria)).ToList();
+            }
+
+            return result;
+        }
+
+        protected virtual Notification[] GetTransientNotifications(NotificationSearchCriteria criteria)
+        {
+            var transientNotificationsQuery = AbstractTypeFactory<Notification>.AllTypeInfos.Select(x =>
+            {
+                return AbstractTypeFactory<Notification>.TryCreateInstance(x.Type.Name);
+            })
+                                                                              .OfType<Notification>().AsQueryable();
+            if (!string.IsNullOrEmpty(criteria.NotificationType))
+            {
+                transientNotificationsQuery = transientNotificationsQuery.Where(x => x.Type.EqualsInvariant(criteria.NotificationType)
+                                                                                    || x.Alias.EqualsInvariant(criteria.NotificationType));
+            }
+
+            transientNotificationsQuery = transientNotificationsQuery.Where(x => !x.Kind.EqualsInvariant(x.Type));
+
+            var transientNotifications = transientNotificationsQuery.Skip(criteria.Skip).Take(criteria.Take).ToArray();
+            foreach (var transientNotification in transientNotifications)
+            {
+                transientNotification.ReduceDetails(criteria.ResponseGroup);
+            }
+
+            return transientNotifications;
+        }
+
+        protected virtual async Task<Notification[]> GetActiveNotifications(NotificationSearchCriteria criteria)
+        {
+            var result = Array.Empty<Notification>();
             var sortInfos = BuildSortExpression(criteria);
-            var tmpSkip = 0;
-            var tmpTake = 0;
 
             using (var repository = _repositoryFactory())
             {
                 var query = BuildQuery(repository, criteria, sortInfos);
 
-                result.TotalCount = await query.CountAsync();
                 if (criteria.Take > 0)
                 {
-                    var notificationIds = await query.OrderBySortInfos(sortInfos).ThenBy(x=>x.Id)
+                    var notificationIds = await query.OrderBySortInfos(sortInfos).ThenBy(x => x.Id)
                                                      .Select(x => x.Id)
                                                      .Skip(criteria.Skip).Take(criteria.Take)
                                                      .ToArrayAsync();
                     var unorderedResults = await _notificationService.GetByIdsAsync(notificationIds, criteria.ResponseGroup);
-                    result.Results = unorderedResults.OrderBy(x => Array.IndexOf(notificationIds, x.Id)).ToList();
+                    result = unorderedResults.OrderBy(x => Array.IndexOf(notificationIds, x.Id)).ToArray();
```

Comment:
Seems you have broken multiple data source pagination with transient and persisted notifications.
Could you return back the original code?
In additional need to apply to `ReduceDetails(criteria.ResponseGroup);` for all notifications not only for transient as it made before.